testing/test_collect_imports.py

```diff
@@ -138,31 +151,83 @@ def test_collect_function_imports_enabled(pytester: Pytester) -> None:
+    # valid tests and `test_function()` will fail.
+    assert len(items_collected) == 3
+    for x in items_collected:
+        assert x.item._getobj().__name__ in [
+            "test_important",
+            "test_bar",
+            "test_function",
+        ]
 
 
-def test_collect_function_imports_disabled(pytester: Pytester) -> None:
+def test_behaviour_without_testpaths_set_and_false(pytester: Pytester) -> None:
+    # Make sure `collect_imported_tests` has no dependence on `testpaths`
     pytester.makeini("""
         [pytest]
-        # testpaths = "tests"
         collect_imported_tests = false
     """)
 
     run_import_functions_test(pytester, passed=1, errors=0, failed=0)
+    reprec = pytester.inline_run()
+    items_collected = reprec.getcalls("pytest_itemcollected")
+    assert len(items_collected) == 1
+    for x in items_collected:
+        assert x.item._getobj().__name__ == "test_important"
 
 
-def test_behaviour_without_testpaths_set_and_false(pytester: Pytester) -> None:
+def test_behaviour_without_testpaths_set_and_true(pytester: Pytester) -> None:
+    # Make sure `collect_imported_tests` has no dependence on `testpaths`
     pytester.makeini("""
         [pytest]
-        collect_imported_tests = false
+        collect_imported_tests = true
     """)
 
-    run_import_functions_test(pytester, passed=1, errors=0, failed=0)
+    run_import_functions_test(pytester, passed=2, errors=0, failed=1)
+    reprec = pytester.inline_run()
+    items_collected = reprec.getcalls("pytest_itemcollected")
+    assert len(items_collected) == 3
 
 
-def test_behaviour_without_testpaths_set_and_true(pytester: Pytester) -> None:
+def test_hook_behaviour_when_collect_off(pytester: Pytester) -> None:
     pytester.makeini("""
         [pytest]
-        collect_imported_tests = true
+        collect_imported_tests = false
     """)
 
-    run_import_functions_test(pytester, passed=2, errors=0, failed=1)
+    run_import_functions_test(pytester, passed=1, errors=0, failed=0)
+    reprec = pytester.inline_run()
+
+    # reports = reprec.getreports("pytest_collectreport")
+    items_collected = reprec.getcalls("pytest_itemcollected")
+    modified = reprec.getcalls("pytest_collection_modifyitems")
+
+    # print("Reports: ----------------")
+    # print(reports)
+    # for r in reports:
+    #     print(r)
+
+    # TODO this is want I want, I think....
+    # <CollectReport '' lenresult=1 outcome='passed'>
+    # <CollectReport 'tests/foo_test.py::TestDomain' lenresult=1 outcome='passed'>
+    # <CollectReport 'tests/foo_test.py' lenresult=1 outcome='passed'>
+    # <CollectReport 'tests' lenresult=1 outcome='passed'>
+    # <CollectReport '.' lenresult=1 outcome='passed'>
+
+    # TODO
+    # assert(reports.outcome == "passed")
+    # assert(len(reports.result) == 1)
+
+    # print("Items collected: ----------------")
+    # print(items_collected)
+    # print("Modified : ----------------")
+
```

Comment:
Im not completely sure what  is the correct behaviour yet, I'll try and figure it out. WIP and all.